src/api/Feed.js

```diff
@@ -95,6 +103,75 @@ const getItemWithPendingReply = <T: { replies?: Array<Comment> }>(item: T, reply
     return { replies: [...replies, reply], ...rest };
 };
 
+const parseFileActivitiesResponseForFeed = (response?: { entries: FileActivity[] }) => {
+    if (!response || !response.entries || !response.entries.length) {
+        return [];
+    }
+
+    const data = response.entries;
+    const parsedData = [];
+
+    data.map(item => {
+        if (!item.source) {
+            return item;
+        }
+
+        const source = { ...item.source };
+        // $FlowFixMe
+        const commentItem: Comment = source[FILE_ACTIVITY_TYPE_COMMENT];
+        // $FlowFixMe
+        const taskItem: Task = source[FILE_ACTIVITY_TYPE_TASK];
+
+        switch (item.type) {
+            case FILE_ACTIVITY_TYPE_TASK:
+                // UAA follows a lowercased enum naming convention, convert to uppercase to align with task api
+                if (!!taskItem.assigned_to && !!taskItem.assigned_to.entries) {
+                    // $FlowFixMe
+                    taskItem.assigned_to.entries.map(entry => {
+                        entry.role = entry.role.toUpperCase();
+                        entry.status = entry.status.toUpperCase();
+
+                        return entry;
```

Comment:
We're still mutating the underlying `entry` object. Can we return a new object using destructuring/rest operators?

Reply:
Would it not be modifying just the cloned source object here?: `const source = { ...item.source };`